libleptongui/src/i_callbacks.c

```diff
@@ -49,7 +34,8 @@
  *  since there is a call: widget = NULL, data = 0 (will be w_current hack)
  *  \todo This should be renamed to page_new perhaps...
  */
-DEFINE_I_CALLBACK(file_new)
+void
+i_callback_file_new (GtkWidget *widget, gpointer data)
```

Comment:
What is the `widget` parameter in all `i_callback_XXX()` functions?
It doesn't used anywhere.

Reply:
While it is not used in Scheme, it may be used when you want to run the callbacks from C.   Some callbacks are indeed used in C, in the toolbar button creation code, and I've unified some of them and removed duplicates.  Since they are connected to glib signals such as "clicked" or "toggled", we have to use the standard pattern for those signals.  So the `widget` parameters for those callbacks are buttons utilizing the signals.  I tried to make all i_callbacks callable from both Scheme and C, though did not managed to do this for some of them yet since it requires more work on resolving C and Scheme code interdependence.